render/render.go

```diff
@@ -109,97 +108,53 @@ func (r *Renderer) DemoKeypress(str string) {
 // ListenToQueue is a blocking function that processes data sent to the RenderQueue
 func (r *Renderer) ListenToQueue() {
 	for {
-		var diff strings.Builder
-		for y := 0; y <= r.h; y++ {
-			for x := 0; x < r.w; x++ {
-				r.writingMutex.Lock()
-				current := r.currentScreen[y][x]
-				pending := r.pendingScreen[y][x]
-				if current != pending {
-					r.currentScreen[y][x] = pending
-
-					if !pending.PrevWide {
-						newCursor := ecma48.Cursor{
-							X: x, Y: y, Style: pending.Style,
-						}
-
-						delta := deltaMarkup(r.drawingCursor, newCursor)
-						diff.WriteString(delta)
-						diff.WriteString(string(pending.Rune))
-
-						if pending.IsWide {
-							newCursor.X += 2
-						} else {
-							newCursor.X++
-						}
-
-						r.drawingCursor = newCursor
-					}
+		for {
+			fullyWritten := true
+			var diff strings.Builder
+			for y := 0; y <= r.h; y++ {
```

Comment:
Is height supposed to be included in the loop?

Reply:
I'm not sure what you're asking. The `diff.Len() > 4000` could go either inside `for y` or both `for y for x`. I don't know which it should be :-/

Reply:
The length is now inside both loops. Being checked more often gives us more control over chunk length

Reply:
Oh! Sorry, I was just wondering if it should be `y < r.h` instead of `y <= r.h`

Reply:
Oh, I'm really not sure. The meaning of width and height in the window manager (i.e. whether they're inclusive) is not documented. Thanks for pointing that out; I just added a note to https://github.com/aaronjanse/3mux/issues/83